src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/GeoServerInterceptorService.java

```diff
@@ -0,0 +1,611 @@
+
+		// intercept the response (if needed)
+		Response interceptedResponse = ogcMessageDistributor
+				.distributeToResponeInterceptor(response, message);
+
+		// finally filter the white-listed response headers
+		// TODO: Move to global proxy class
+		interceptedResponse.setHeaders(
+				getHeadersToForward(interceptedResponse.getHeaders()));
+
+		return interceptedResponse;
+	}
+
+	/**
+	 *
+	 * @param mutableRequest
+	 * @return
+	 * @throws InterceptorException
+	 */
+	private OgcMessage getOgcRequest(MutableHttpServletRequest mutableRequest)
+			throws InterceptorException {
+
+		OgcMessage ogcRequest = new OgcMessage();
+
+		String requestService = getRequestParameterValue(
+				mutableRequest, OgcNaming.PARAMETER_SERVICE);
+		String requestOperation = getRequestParameterValue(
+				mutableRequest, OgcNaming.PARAMETER_OPERATION);
+		String requestLayer = getRequestParameterValue(
+				mutableRequest, OgcNaming.PARAMETER_ENDPOINT);
+		InterceptorRule mostSpecificRequestRule = getMostSpecificRule(requestService,
+				requestOperation, requestLayer, "REQUEST");
+		InterceptorRule mostSpecificResponseRule = getMostSpecificRule(requestService,
+				requestOperation, requestLayer, "RESPONSE");
+
+		if (StringUtils.isNoneEmpty(requestService)) {
+			ogcRequest.setService(requestService);
+		} else {
+			LOG.debug("No service found.");
+		}
+
+		if (StringUtils.isNoneEmpty(requestOperation)) {
+			ogcRequest.setOperation(requestOperation);
+		} else {
+			LOG.debug("No operation found.");
+		}
+
+		if (StringUtils.isNoneEmpty(requestLayer)) {
+			ogcRequest.setEndPoint(requestLayer);
+		} else {
+			LOG.debug("No endPoint found.");
+		}
+
+		if (mostSpecificRequestRule != null) {
+			ogcRequest.setRequestRule(mostSpecificRequestRule.getRule());
+		} else {
+			LOG.debug("No interceptor rule found for the request.");
+		}
+
+		if (mostSpecificResponseRule != null) {
+			ogcRequest.setResponseRule(mostSpecificResponseRule.getRule());
+		} else {
+			LOG.debug("No interceptor rule found for the response.");
+		}
+
+		if (StringUtils.isEmpty(requestService) &&
+				StringUtils.isEmpty(requestOperation) &&
+				StringUtils.isEmpty(requestLayer)) {
+			throw new InterceptorException("Couldn't find all required OGC " +
+					"parameters (SERVICE, REQUEST, LAYER). Please check the " +
+					"validity of the request.");
+		}
+
+		return ogcRequest;
+	}
+
+	/**
+	 *
+	 * @param requestService
+	 * @param requestOperation
+	 * @param requestLayer
+	 * @return
+	 */
+	private InterceptorRule getMostSpecificRule(String requestService,
+			String requestOperation, String requestEndPoint, String ruleEvent) {
+
+		List<InterceptorRule> result;
+
+		Map<String, String> filterMap = new HashMap<String, String>();
+
+		filterMap.put("event", ruleEvent);
+
+		filterMap.put("service", requestService);
+		filterMap.put("operation", requestOperation);
+		filterMap.put("endPoint", requestEndPoint);
+
+		result = this.interceptorRuleService.findSpecificRule(filterMap);
```

Comment:
This will result in a database query. And in the worst case it will result in up to 4 queries, which is of course not optimal for every HTTP-request. We have been talking this through and I think this is OK for the moment. A future refactoring may make this more efficient, either by combining requests, reading the complete table (if possible) into memory or some other clever approach. 

Long story short: OK for now.